config/mapflags.go

```diff
@@ -0,0 +1,69 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+// mapFlags are generic string key-value pair flags.
+// Use when option keys are not predetermined.
+type mapFlags struct {
+	values map[string]string
+}
+
+const formatErrorString = "invalid map key-value pair, expected format key=value but got: '%v'"
+
+func newMapFlags() *mapFlags {
+	return &mapFlags{
+		values: make(map[string]string),
+	}
+}
+
+func (m mapFlags) String() string {
```

Comment:
should be pointer receiver as elsewhere

Reply:
Fixed

Reply:
@AlexanderYastrebov we can let it like this but this is not true for String() it's normal to have value receiver, see also https://go-talks.appspot.com/github.com/szuecs/slides/go-class/go-class-kickstarter.slide#31 https://go-talks.appspot.com/github.com/szuecs/slides/go-class/go-class-kickstarter.slide#32